presto-tpch/src/main/java/com/facebook/presto/util/Optionals.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.util;
+
+import java.util.Optional;
+import java.util.function.BiFunction;
+import java.util.function.BinaryOperator;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class Optionals
+{
+    private Optionals() {}
+
+    public static <T> Optional<T> checkPresent(Optional<T> optional)
+    {
+        checkArgument(optional.isPresent(), "Expected a present optional, got empty()");
+        return optional;
+    }
+
+    public static <S, T, R> Optional<R> withBoth(Optional<? extends S> left, Optional<? extends T> right, BiFunction<S, T, R> binaryFunction)
+    {
+        return left.flatMap(l -> right.map(r -> binaryFunction.apply(l, r)));
+    }
+
+    public static  <T> Optional<T> combine(Optional<T> left, Optional<T> right, BinaryOperator<T> combiner)
```

Comment:
it's used 4 times, so no to inlining :) It's used in one class, true. But it's usage-site agnostic and I'd rather leave it as a util and see it used in one place only, than leave it in StatisticsEstiamtor and give no one a chance to find it and avoid duplicating it.